### PR TITLE
Stripe improve logging and error reporting

### PR DESCRIPTION
**Pull request by aivils**

### CODE REVIEW

File: app/services/stripe_service/api/payments.rb [hunk 1]
```diff
@@ -129,6 +135,8 @@ def payment_details(tx)
       end
 
       def payout(tx)
+        report = StripeService::Report.new(tx: tx)
+        report.capture_charge_start
```
Comment (bladealslayer) [inline app/services/stripe_service/api/payments.rb]: This report is in the wrong method. report.capture_charge should be used in the `capture` method above. Here the call is for creating payout. Add also reporting for creating payout in the `Report` class.

For payouts we should log:

before:

```json
{
  "stripe_op": "create_payout", 
  "event": "stripe_call",
  "transaction_id": 1234,
  "stripe_payment_id": 123,
  "stripe_seller_id": "acc_foobar",
  "stripe_charge_id": "ch_foobar"
}
```

after (success):

```json
{
  "stripe_op": "create_payout",
  "event": "stripe_call_succeeded",
  "transaction_id": 1234,
  "stripe_payment_id": 123,
  "stripe_seller_id": "acc_foobar",
  "stripe_charge_id": "ch_foobar",
  "stripe_payout_id": "po_foobar"
}
```

after (failure):

```json
{
  "stripe_op": "create_payout",
  "event": "stripe_call_failed",
  "transaction_id": 1234,
  "stripe_payment_id": 123,
  "stripe_seller_id": "acc_foobar",
  "stripe_charge_id": "ch_foobar",
  "stripe_error": {
    "message": "the 'message' from Stripe API response, if any",
    "code": "the code in stripe api response if any",
    "decline_code": "the decline_code if any"
  }
}
```

File: app/services/stripe_service/report.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    result = create_charge.merge({
+      "event": "stripe_call",
+    })
+    logger.info('create_charge_start', nil, result)
+    result
+  end
+
+  def create_charge_success
+    result = create_charge.merge({
+      "event": "stripe_call",
```
Comment (bladealslayer) [inline app/services/stripe_service/report.rb]: `stripe_call_succeeded`

File: app/services/stripe_service/report.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      "stripe_charge_id": stripe_payment.stripe_charge_id,
+      "stripe_charge_state": "success"
+    })
+    logger.info('create_charge_success', nil, result)
+    result
+  end
+
+  def create_charge_failed
+    result = create_charge.merge({
+      "event": "stripe_call",
```
Comment (bladealslayer) [inline app/services/stripe_service/report.rb]: `stripe_call_failed`

**Comment by aivils** [comment]
@bladealslayer 
Changes applied.

File: app/services/stripe_service/report.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    logger.info('create_payout_start', nil, result)
+    result
+  end
+
+  def create_payout_success
+    result = create_payout.merge({
+      "event": "stripe_call_succeeded",
+      "stripe_payment_id": stripe_payment.id,
+      "stripe_charge_id": stripe_payment.stripe_charge_id,
+      "stripe_transfer_id": stripe_payment.stripe_transfer_id
```
Comment (bladealslayer) [inline app/services/stripe_service/report.rb]: @aivils for some reason this was logged as `nil`, even though the payment does have the column updated. Any idea why?

Comment (bladealslayer) [reply app/services/stripe_service/report.rb]: Right, need for `.reload`... Thanks for fixing @aivils 👍

**Pull request merged**